src/components/Export.tsx

```diff
@@ -365,6 +366,49 @@ const ExportDialog: ExportDialogComponent = ({
     }
   };
 
+  const handlePdfExport = async (
+    files: {
+      title: string;
+      content: string;
+    }[]
+  ) => {
+    const preparedFiles = files.map((f) => ({
+      title: JSON.stringify(f.title),
+      content: JSON.stringify(f.content),
+    }));
+    const domain =
+      getNodeEnv() === "development"
+        ? "http://localhost:3003"
+        : "https://api.samepage.network";
+    const response = await apiPost({
+      domain,
+      path: "apps/query-builder/pdf",
+      data: {
+        files: preparedFiles,
+      },
+    }).catch(() => {
+      setError("Failed to export files.");
+    });
+
+    if (response) {
+      const parsedResponse = JSON.parse(response.data);
+      const base64ToBlob = (
+        base64: string,
+        type = "application/octet-stream"
+      ) => {
+        const binaryString = window.atob(base64);
+        const len = binaryString.length;
+        const bytes = new Uint8Array(len);
+        for (let i = 0; i < len; i++) {
+          bytes[i] = binaryString.charCodeAt(i);
+        }
+        return new Blob([bytes], { type: type });
+      };
+      const blob = base64ToBlob(parsedResponse.body, "application/zip");
+      saveAs(blob, `${filename}.zip`);
+      onClose();
```

Comment:
this conversion looks expensive - here's a possible alternative:
- PDF conversion is done and uploaded to a public link `.zip` in samepage's backend
- we download straight from that link here

Reply:
Do we already have a utility for downloading files from samepage's backend (s3) from a frontend extension? (I didn't see one)

Or do we need to create a lambda to generate a signed url?  Or something else?

Reply:
The expectation is that we don't need a lambda - as long as the path matches one of these origin paths that it should download: https://github.com/samepage-network/samepage.network/blob/main/app/server/index.ts

Reply:
Sorry, I don't follow.

Currently I'm using `uploadFileContent` to upload to s3 @ `data/pdfs`.
I'm not sure how to use `app/server/index` to retrieve a file uploaded to there.

Would I need to add `data/pdfs` to the `originPaths` in `app/server/index`?
Then create something like `/api/pdf/get.ts` to invoke `app/server/index`'s handler?

Reply:
I tried adding `/^\/data\/pdfs\/.+$/,` to `originPaths` then retrieving in query builder with:

```typescript
apiGet<ArrayBuffer>({
        domain: `https://api.samepage.network`,
        path: `data/pdfs/${file}`
        buffer: true,
      });
```
      
But this results in a CORS error.
      
I have a feeling that `https://api.samepage.network` is incorrect
Is there a separate url for retrieving from s3?

Reply:
ah it would just be from `https://samepage.network` - the `originPaths` within `app/server/index` define all the paths where we simply just redirect the request to retrieve the content from S3.

As a general rule, we should try to avoid putting any "public assets" within the `data` directory, and instead, put the pdfs in likely a new top level `pdfs`

Reply:
Excellent.

After adding `/^\/pdfs\/.*/`, `localhost` kept giving me `CORS` errors.  So I used `/videos` to test.
Just pinging you on this because there are some test uploads in that route, but I don't believe I have access to delete them.